src/search/tasks/negated_axioms_task.cc

```diff
@@ -0,0 +1,358 @@
+        bool same_var_occurs_already = false;
+        for (const FactPair &f : hitting_set) {
+            if (f.var == elem.var) {
+                same_var_occurs_already = true;
+                break;
+            }
+        }
+        if (same_var_occurs_already) {
+            continue;
+        }
+
+        set<FactPair> chosen_new(hitting_set);
+        chosen_new.insert(elem);
+        collect_non_dominated_hitting_sets_recursively(
+            conditions_as_cnf, index + 1, chosen_new, results);
+    }
+}
+
+
+int NegatedAxiomsTask::get_operator_cost(int index, bool is_axiom) const {
+    if (!is_axiom || index < negated_axioms_start_index) {
+        return parent->get_operator_cost(index, is_axiom);
+    }
+
+    return 0;
+}
+
+string NegatedAxiomsTask::get_operator_name(int index, bool is_axiom) const {
+    if (!is_axiom || index < negated_axioms_start_index) {
+        return parent->get_operator_name(index, is_axiom);
+    }
+
+    return "<axiom>";
+}
+
+int NegatedAxiomsTask::get_num_operator_preconditions(int index, bool is_axiom) const {
+    if (!is_axiom || index < negated_axioms_start_index) {
+        return parent->get_num_operator_preconditions(index, is_axiom);
+    }
+
+    return negated_axioms[index - negated_axioms_start_index].condition.size();
+}
+
+FactPair NegatedAxiomsTask::get_operator_precondition(
+    int op_index, int fact_index, bool is_axiom) const {
+    if (!is_axiom || (op_index < negated_axioms_start_index)) {
+        return parent->get_operator_precondition(op_index, fact_index, is_axiom);
+    }
+    return negated_axioms[op_index - negated_axioms_start_index].condition[fact_index];
+}
+
+int NegatedAxiomsTask::get_num_operator_effects(int op_index, bool is_axiom) const {
+    if (!is_axiom || op_index < negated_axioms_start_index) {
+        return parent->get_num_operator_effects(op_index, is_axiom);
+    }
+
+    return 1;
+}
+
+int NegatedAxiomsTask::get_num_operator_effect_conditions(
+    int op_index, int eff_index, bool is_axiom) const {
+    if (!is_axiom || op_index < negated_axioms_start_index) {
+        return parent->get_num_operator_effect_conditions(op_index, eff_index, is_axiom);
+    }
+
+    return 0;
+}
+
+FactPair NegatedAxiomsTask::get_operator_effect_condition(
+    int op_index, int eff_index, int cond_index, bool is_axiom) const {
+    assert(!is_axiom || op_index < negated_axioms_start_index);
+    return parent->get_operator_effect_condition(op_index, eff_index, cond_index, is_axiom);
+}
+
+FactPair NegatedAxiomsTask::get_operator_effect(
+    int op_index, int eff_index, bool is_axiom) const {
+    if (!is_axiom || op_index < negated_axioms_start_index) {
+        return parent->get_operator_effect(op_index, eff_index, is_axiom);
+    }
+
+    assert(eff_index == 0);
+    return negated_axioms[op_index - negated_axioms_start_index].head;
+}
+
+int NegatedAxiomsTask::convert_operator_index_to_parent(int index) const {
+    return index;
```

Comment:
I don't think this should be overwritten at all.

Reply:
I removed the method.